### PR TITLE
feature: metrics 0.14 & ability to flush to disk less

### PR DESCRIPTION
**Pull request by jfro**
- Allows less frequent writing to disk by doing a single transaction batch on a flush timeout interval
- Upgrades to `metrics` 0.14 which has float support & simpler API

## Todo

- [x] Cleanup, document etc

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,17 +10,14 @@ license = "MIT OR Apache-2.0"
...
+# this seems to let us force bundled sqlite in diesel
+libsqlite3-sys = { version = "0.18", features = ["bundled"] }
+metrics = { version = "0.14" }
+thiserror = "1.0"
+log = "0.4"
 
 [dev-dependencies]
-metrics-runtime = { version = "0.13.1", default-features = false }
-metrics = "0.12.1"
+metrics = "0.14"
```
Comment (jesdazrez) [inline Cargo.toml]: Given that we now have `metrics` as a regular dep, could we remove this?

File: src/lib.rs [hunk 2]
```diff
@@ -60,218 +66,118 @@ fn setup_db<P: AsRef<Path>>(path: P) -> Result<SqliteConnection> {
...
+        let mut queue = VecDeque::with_capacity(FLUSH_QUEUE_LIMIT);
+        let mut last_flush = Instant::now();
+        fn flush(
+            queue: &mut VecDeque<NewMetric>,
+            db: &SqliteConnection,
+        ) -> Result<(), diesel::result::Error> {
+            use crate::schema::metrics::dsl::metrics;
+            trace!("Flushing {} records", queue.len());
+            db.transaction::<_, diesel::result::Error, _>(|| {
+                for rec in queue.drain(0..) {
```
Comment (jesdazrez) [inline src/lib.rs]: This could be simplified to just `..`

File: src/lib.rs [hunk 3]
```diff
@@ -60,218 +66,118 @@ fn setup_db<P: AsRef<Path>>(path: P) -> Result<SqliteConnection> {
...
-            for (name, value) in values {
-                self.contents
-                    .insert(format!("{}.{}", levels, name), value as i64);
+        loop {
+            let (should_flush, should_exit) = match receiver.recv_timeout(flush_duration) {
+                Ok(Event::Stop) => {
+                    info!("Stopping SQLiteExporter worker");
+                    (true, true)
+                }
+                // Ok(Event::Flush) => (true, false),
```
Comment (jesdazrez) [inline src/lib.rs:111]: Is `Event::Flush` not supported anymore?

Comment (jfro) [reply src/lib.rs:111]: I just didn't have use for it after writing it. I think it'll be useful if we want to expose a manual flush call on the exporter. common use currently is to install exporter & you lose access to it so... future feature work if desired.

File: Cargo.toml (hunk 1 above)
Comment (jfro) [reply Cargo.toml]: oops, yup!

**Review by jesdazrez — APPROVED** [review]

**Pull request merged**